docs/README.md

```diff
@@ -1,3 +1,201 @@
+```csharp
+    using System;
+    using System.Runtime.InteropServices;
+    using SwiftRuntimeLibrary;
+    using SwiftRuntimeLibrary.SwiftMarshal;
+    
+    namespace NewClassCompilerTests
+    {
+        [SwiftStruct("libNewClassCompilerTests.dylib",
+            "$s21NewClassCompilerTests6BarIntVMn", 
+            "$s21NewClassCompilerTests6BarIntVN", "")]
+        public class BarInt : ISwiftStruct
+        {
+            public BarInt(nint x)
+            {
+            }
+            internal BarInt(SwiftNominalCtorArgument unused)
+            {
+            }
+            public static SwiftMetatype GetSwiftMetatype()
+            {
+            }
+            public void Dispose()
+            {
+            }
+            void Dispose(bool disposing)
+            {
+            }
+            ~BarInt()
+            {
+            }
+            public byte[] SwiftData
+            {
+                get; set;
+            }
+            public nint X
+            {
+                get { }
+                set { }
+            }
+        }
+    }
+```
+
+There is a payload `SwiftData` along with two constructors. The first maps onto the `init` method inside the swift class. The second is an internal constructor that gets used to define an uninitialized type. This constructor gets used by the marshaler when a type needs to be allocated before it gets used, for example, as a return value because Swift semantics don’t allow to explicitly have variables in an uninitialized state. 
+
+All properties bound in C# access the value through accessor functions. If a Swift property or a Swift method mutates the value, the contents of the C# `SwiftData` property will get changed as well.
+
+### Enums
+
+Swift enums are projected into C# as `IDisposable` classes, implementing `ISwiftEnum`, in similar manner as structs.
+
+Given this Swift enum:
+```swift
+    public enum FooECTIA {
+        case a(Int)
+        case b(Double)
+    }
+```
+
+The projection tooling will generate the following C#, with function bodies left empty for simplicity:
+```csharp
+    using System;
+    using System.Runtime.InteropServices;
+    using SwiftRuntimeLibrary;
+    using SwiftRuntimeLibrary.SwiftMarshal;
+    
+    namespace NewClassCompilerTests
+    {
+        public enum FooECTIACases
+        {
+            A, B
+        }
+        [SwiftEnumType("libNewClassCompilerTests.dylib",
+            "$s21NewClassCompilerTests8FooECTIAOMn", 
+            "$s21NewClassCompilerTests8FooECTIAON", "")]
+        public class FooECTIA : ISwiftEnum
+        {
+            public void Dispose()
+            {
+            }
+            void Dispose(bool disposing)
+            {
+            }
+            ~FooECTIA()
+            {
+            }
+            public static FooECTIA NewA(nint value0)
+            {
+            }
+            public static FooECTIA NewB(double value0)
+            {
+            }
```

Comment:
The [naming conventions](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods) says to use verbs for methods:

```suggestion
            public static FooECTIA CreateA(nint value0)
            {
            }
            public static FooECTIA CreateB(double value0)
            {
            }
```

Reply:
Although this might be a potential alternative design?

```csharp
public class FooECTIA : ISwiftEnum {
    public FooECTIA (FooECTIACases case, nint value) {}
    public FooECTIA (FooECTIACases case, double value) {}
}
```

This could be simplified to something like this if the value type defines the case as well:

```csharp
public class FooECTIA : ISwiftEnum {
    public FooECTIA (nint value) {}
    public FooECTIA (double value) {}
}
```

Reply:
> The [naming conventions](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods) says to use verbs for methods:

Added to the implementation plan. 



> ```cs
> public class FooECTIA : ISwiftEnum {
>     public FooECTIA (FooECTIACases case, nint value) {}
>     public FooECTIA (FooECTIACases case, double value) {}
> }
> ```

Your suggestion is sensible and appears to simplify usage but It is hard to make a decision at this stage. The existing factory-like approach is easier for maintenance and further extensibility. I will add it to the list of tasks and we can consider implementing it at some later stage.